File: frontend/src/api/audit.ts
import api from './client'

export interface AuditRecord {
  id: number
  username: string
  tool_name: string
  ai_suggested_args: Record<string, unknown>
  user_edited_args: Record<string, unknown> | null
  result_status: 'approved' | 'rejected'
  session_id: string
  thread_id: string
  timestamp: string
}

export interface AuditFilters {
  username?: string
  tool_name?: string
  date_from?: string
  date_to?: string
  limit?: number
  offset?: number
}

export async function fetchAuditLog(filters: AuditFilters = {}): Promise<{ records: AuditRecord[]; count: number }> {
  const { data } = await api.get('/audit', { params: filters })
  return data
}

File: backend/app/api/v1/audit.py
from datetime import datetime
from typing import Optional

from fastapi import APIRouter, Query

from app.services.audit import get_audit_records

router = APIRouter(prefix="/audit", tags=["audit"])


@router.get("")
def list_audit(
    username: Optional[str] = Query(None),
    tool_name: Optional[str] = Query(None),
    date_from: Optional[datetime] = Query(None),
    date_to: Optional[datetime] = Query(None),
    limit: int = Query(100, ge=1, le=500),
    offset: int = Query(0, ge=0),
):
    records = get_audit_records(
        username=username,
        tool_name=tool_name,
        date_from=date_from,
        date_to=date_to,
        limit=limit,
        offset=offset,
    )
    return {"records": records, "count": len(records)}
